ingestors/kafka/src/main/scala/hydra/kafka/endpoints/BootstrapEndpoint.scala

```diff
@@ -40,41 +40,44 @@ class BootstrapEndpoint(override val system:ActorSystem) extends RouteSupport
 
   private implicit val timeout = Timeout(10.seconds)
 
-  override val route: Route = cors(settings) {
-    pathPrefix("streams") {
-      pathEndOrSingleSlash {
-        post {
-          requestEntityPresent {
-            entity(as[TopicMetadataRequest]) { topicMetadataRequest =>
-              onComplete(
-                bootstrapActor ? InitiateTopicBootstrap(topicMetadataRequest)
-              ) {
-                case Success(message) =>
-                  message match {
-
-                    case BootstrapSuccess(metadata) =>
-                      complete(StatusCodes.OK, toResource(metadata))
+  private def topicRoute =
+    pathEndOrSingleSlash {
+    post {
+      requestEntityPresent {
+        entity(as[TopicMetadataRequest]) { topicMetadataRequest =>
+          onComplete(
+            bootstrapActor ? InitiateTopicBootstrap(topicMetadataRequest)
+          ) {
+            case Success(message) =>
+              message match {
 
-                    case BootstrapFailure(reasons) =>
-                      complete(StatusCodes.BadRequest, reasons)
+                case BootstrapSuccess(metadata) =>
+                  complete(StatusCodes.OK, toResource(metadata))
 
-                    case e: Exception =>
-                      log.error("Unexpected error in TopicBootstrapActor", e)
-                      complete(StatusCodes.InternalServerError, e.getMessage)
-                  }
+                case BootstrapFailure(reasons) =>
+                  complete(StatusCodes.BadRequest, reasons)
 
-                case Failure(ex) =>
-                  log.error("Unexpected error in BootstrapEndpoint", ex)
-                  complete(StatusCodes.InternalServerError, ex.getMessage)
+                case e: Exception =>
+                  log.error("Unexpected error in TopicBootstrapActor", e)
+                  complete(StatusCodes.InternalServerError, e.getMessage)
```

Comment:
Is this actually reachable? We are matching on `message` which it looks like is either a `BootstrapSuccess` or a `BootstrapFailure`. `Success(e: Exception)` would be an antipattern as `Failure` should be used to contain exceptions (and I'm guessing that is the case)

Reply:
This is in the original BootstrapEndpoint file, so I moved it over as I figured it had been added for a reason. I have not looked into it however.

Reply:
I am not seeing it in the diff there, can you post a link to the original for me when you get a chance?

Reply:
Line 61 to the left.

Reply:
Ah thank you, I totally missed that. Feel free to disregard this.